test/main.js

```diff
@@ -0,0 +1,41 @@
+const test = require('ava');
+const got = require('got');
+
+const plugins = require('../site/plugins.json');
+
+const STRING_ATTRIBUTES = ['author', 'description', 'name', 'package', 'repo', 'status', 'version'];
+const OPTIONAL_ATTRIBUTES = ['status'];
+const ATTRIBUTES = [...STRING_ATTRIBUTES];
+const ENUMS = {
+  status: ['DEACTIVATED', undefined],
+};
+
+plugins.forEach((plugin) => {
+  const { package: packageName, repo, version } = plugin;
+
+  Object.entries(plugin).forEach(([attribute, value]) => {
+    test(`Plugin attribute "${attribute}" should have a proper shape: ${packageName}`, (t) => {
+      t.true(ATTRIBUTES.includes(attribute));
+
+      const possibleValues = ENUMS[attribute];
+      t.true(possibleValues === undefined || possibleValues.includes(value));
+
+      if (value === undefined && OPTIONAL_ATTRIBUTES.includes(attribute)) {
+        return;
+      }
+
+      if (STRING_ATTRIBUTES.includes(attribute)) {
+        t.is(typeof value, 'string');
+        t.not(value.trim(), '');
+      }
+    });
+  });
+
+  test(`Plugin package should be published: ${packageName}`, async (t) => {
+    await t.notThrowsAsync(got(`https://www.npmjs.com/package/${packageName}/v/${version}`));
```

Comment:
[sand] Should we use `pacote` for verifying the package?
See https://github.com/netlify/plugins/blob/4c55407c1b51ab232134714ab2414b2d9a54d7e8/functions/npm-diff.js#L86

Reply:
Yes, this is better. Thanks for the suggestion!